detect/region.go

```diff
@@ -0,0 +1,66 @@
+package detect
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/urfave/cli"
+	"gopkg.in/errgo.v1"
+
+	"github.com/Scalingo/cli/config"
+	"github.com/Scalingo/cli/utils"
+
+	"github.com/Scalingo/go-scalingo/v4/debug"
+)
+
+// CurrentRegion extract and returns the region from the Git remote URL
+// If not found it return an empty string
+func CurrentRegion(c *cli.Context, rc *config.RegionsCache) string {
+	if dir, ok := utils.DetectGit(); ok {
+		currentRegion, err := getRegionFromGitRemote(dir, c.GlobalString("remote"), rc)
+		if err != nil {
+			debug.Println(err)
+		}
+		debug.Println("[detect] Region name is", currentRegion)
+		return currentRegion
+	}
+	return ""
+}
+
+// ScalingoRepo searches into the current directory and its parent for a remote
+// named remoteName or scalingo-<remoteName>.
+//
+// It returns the region and an error.
```

Comment:
I expect "region" to be something with the type `scalingo.Region`.

```suggestion
// It returns the region name and an error.
```

Reply:
You are right, it could be misleading, I will replace region with "region name"